### DISCUSSION TITLE
contextualSpellCheck module cannot be loaded

### DISCUSSION THREAD

**Discussion by Abe410** [explosion/spaCy#10610] [Help: Installation] [answered]
Hi guys,

By using the following resource, I am trying to install the spell check library:

https://spacy.io/universe/project/contextualSpellCheck

But after doing the pip install and then importing cpntextualSpellCheck, I get an error "No module named 'contextualSpellCheck'".

Any advice?

Thanks

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @Abe410 ,

Just to sanity-check, did `contextualSpellCheck` install without any problems? You can refer to the installation logs to be sure. One of its dependencies, `editdistance`, requires `wheel` to be installed, so you might want to do something like:

```
pip install wheel  # to solve some setup problems
pip install contextualSpellCheck
```

Another thing to check is your environment, are you importing `contextualSpellCheck` in the same environment where you installed it? You can avoid env problems by [creating virtual environments and the like](https://docs.python.org/3/library/venv.html#module-venv).

**Reply by Abe410** [reply]
Hi

When installing wheel, it gives the message that "Requirement already satisfied".

But when I install contextualSpellCheck, the following is a message I get:

`Building wheels for collected packages: editdistance`
  `Building wheel for editdistance (setup.py): started`
 ` Building wheel for editdistance (setup.py): finished with status 'error'`
  `Running setup.py clean for editdistance`
`Failed to build editdistance`
`Installing collected packages: editdistance, contextualSpellCheck`
    `Running setup.py install for editdistance: started`
    `Running setup.py install for editdistance: finished with status 'error'`

Also:

`building 'editdistance.bycython' extension`
  `error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": [https://visualstudio.microsoft.com/visual-cpp-build-tools/]()`
  ----------------------------------------
  `ERROR: Failed building wheel for editdistance`
   ` ERROR: Command errored out with exit status 1:`

**Reply by ljvmiranda921** [reply]
So what's happening is that `editdistance`, a dependency of `contextualSpellCheck`, is not being built properly. Hence, `contextualSpellCheck` is not installed as a result. Perhaps you can follow what was suggested in the traceback? Might also be worth a shot installing via [conda](https://anaconda.org/conda-forge/editdistance) although I haven't personally tried that.

**Reply by Abe410** [reply]
I already have Visual C++ version 14.0 and up, but still no luck.

I'll try installing via conda and let's see if there is any luck then.

Thanks!

**Reply by Abe410** [reply]
Update

I went through the traceback and turns out I wasn't installing the correct Visual C++.

The following resource can be of help to anyone also stuck with the same problem:

https://www.youtube.com/watch?v=rcI1_e38BWs

Thanks alot Miranda!
